4_streamlit_dashboard_for_model_inference.py

```diff
@@ -0,0 +1,122 @@
+# Import python packages
+import streamlit as st
+from snowflake.snowpark.context import get_active_session
+
+# Snowpark for Python
+from snowflake.snowpark import Session
+from snowflake.snowpark.version import VERSION
+from snowflake.snowpark.functions import udf
+import snowflake.snowpark.functions as F
+
+# Snowpark ML
+from snowflake.ml.modeling.xgboost import XGBRegressor
+from snowflake.ml.modeling.model_selection import GridSearchCV
+from snowflake.ml.registry import Registry
+from snowflake.ml._internal.utils import identifier
+
+
+# Write directly to the app
+st.title("Diamond Price Prediction :ring:")
+st.write(
+    """Call your newly deployed diamond prediction model from Snowflake's model registry!
+    """
+)
+
+# Set up the session
+session = get_active_session()
+db = identifier._get_unescaped_name(session.get_current_database())
+schema = identifier._get_unescaped_name(session.get_current_schema())
+
+#Check out available model versions
+st.write("ML Model Version Table")
+model_name = "DIAMONDS_PRICE_PREDICTION"
+native_registry = Registry(session=session, database_name=db, schema_name=schema)
+st.write(native_registry.get_model(model_name).show_versions())
+
+
+#Create fields to specify model inputs
+st.write("Model inputs")
+
+#Model input columns
+input_cols =[
+ 'CARAT',
+ 'DEPTH',
+ 'TABLE_PCT',
+ 'X',
+ 'Y',
+ 'Z',
+ 'CUT_OE',
+ 'COLOR_OE',
+ 'CLARITY_OE',
+ 'COLOR',
+ 'CLARITY',
+ 'PRICE',
```

Comment:
A nit, but PRICE is the label for this model, so doesn't make sense to include as an input?